collection.go

```diff
@@ -67,10 +122,19 @@ func (c *Client) DeleteCollection(ctx context.Context, dbName, colName string,
 	return ErrorNotImplemented
 }
 
-func (c *Client) ReplaceCollection(ctx context.Context, dbName, colName string,
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/replace-a-collection
+func (c *Client) ReplaceCollection(ctx context.Context, dbName string,
```

Comment:
I think we should drop the `*RequestOptions` parameter and put it in the `CollectionReplaceOptions`. 
Background: The `RequestOptions` was an early attempt in modelling the required options but I changed it in favor of the explicit structs for each operation. The rough plan was a) use explicit options for each struct and then b) see if we can factor out common things in a way that doesn't change the api.

Reply:
sidenote: I did not factor out common fields on purpose yet. I did not trust the documentation enough.